crates/ruff_linter/src/rules/pep8_naming/rules/constant_imported_as_non_constant.rs

```diff
@@ -52,6 +65,9 @@ pub(crate) fn constant_imported_as_non_constant(
     stmt: &Stmt,
     ignore_names: &IgnoreNames,
 ) -> Option<Diagnostic> {
+    // Single-character names are ambiguous. It could be a class or a constant.
+    name.chars().nth(1)?;
```

Comment:
But we still want to forbid things like `from foo import C as c` from _one_ of these rules, right? Won't these changes mean that we will no longer see that error?

Reply:
It's not possible to tell if `C` is a class or a constant. So even though `from foo import C as c` might not be advisable, it's not applicable under a rule made for constants.

Reply:
I could add a new rule, `non_lowercase_imported_as_lowercase`? (to compliment the already-existing `lowercase_imported_as_non_lowercase)

Reply:
Would it be possible to preserve the existing behavior for lower-case letters which is that both rules falg it?

Nice catch @AlexWaygood !

Reply:
> Would it be possible to preserve the existing behavior for lower-case letters which is that both rules falg it?

Yeah, that could be an option.

Reply:
I've just pushed a change that makes N811 flags it. I think that rule makes more sense here, and I think it's not a great outcome if one lint error is flagged by two rules. I don't think the situation is common enough to justify its own rule.